### PR TITLE
add option to enable preview for specific files only

### LINKED ISSUE #5079: Generate previews only for files that can have a preview
Only send previews for known mimetypes that can have them and app allows for that.

The logic should be:
1) Check what apps are installed
2) Get list of mimetypes handled by apps
3) Check if for each mime type the app can generate a preview
4) Send the preview (and cache in browser)

### PR DESCRIPTION
**Pull request by fschade**
## Description
Introduce a new config option to configure which file extension get a preview

## Related Issue

- Fixes #5079

## Motivation and Context
reduce unnecessary requests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

### CODE REVIEW

**Review by kulmann — CHANGES_REQUESTED** [review]
Could you add one list item describing this option to the `options` list in `/docs/getting-started.md` please?

File: packages/web-runtime/src/store/config.js [hunk 1]
```diff
@@ -88,6 +89,10 @@ const getters = {
   configuration: state => {
     return state
   },
+  previewFileExtensions: state => {
+    const extensions = state.options.previewFileExtensions
+    return (Array.isArray(extensions) ? extensions : []).filter(Boolean)
```
[the remark this answers is not in the document]
Comment (fschade) [reply packages/web-runtime/src/store/config.js]: done, good catch. Thanks

**Review by kulmann — APPROVED** [review]
Yes, awesome. Tried it locally, works like a charm. 🚀

**Comment by labkode** [comment]
Looks good to me, adding @diocas for another pair of eyes.

**Comment by diocas** [comment]
For now seems ok. 

But I have some concerns (for the future).. Right now the source of truth on which files can be converted is on the thumbnails service ([here](https://github.com/owncloud/ocis/blob/bbcea23e85b705aafee17e0fbf05d67cc961b819/thumbnails/pkg/thumbnail/encoding.go#L59), I believe). I don't know if it will have other extensions added (or even if the service will be extendable by plugins), but it's odd to me that we don't ask it which extensions it can handle, instead of allowing a manual (possibly out-of-sync) configuration.

**Comment by fschade** [comment]
> For now seems ok.
> 
> But I have some concerns (for the future).. Right now the source of truth on which files can be converted is on the thumbnails service ([here](https://github.com/owncloud/ocis/blob/bbcea23e85b705aafee17e0fbf05d67cc961b819/thumbnails/pkg/thumbnail/encoding.go#L59), I believe). I don't know if it will have other extensions added (or even if the service will be extendable by plugins), but it's odd to me that we don't ask it which extensions it can handle, instead of allowing a manual (possibly out-of-sync) configuration.

The plan for the future is that services can announce what type of thumbnails they can create, this pr is the building block for this and only for the moment statically.

**Comment by diocas** [comment]
@fschade perfect then! :)

**Pull request merged**